src/jvmMain/kotlin/io/beatmaps/api/playlist.kt

```diff
@@ -213,36 +226,40 @@ fun Route.playlistRoute() {
         val searchFields = PgConcat(" ", Playlist.name, Playlist.description)
         val searchInfo = parseSearchQuery(it.q, searchFields)
         val actualSortOrder = searchInfo.validateSearchOrder(it.sortOrder)
+        val sortArgs = when (actualSortOrder) {
+            SearchOrder.Relevance -> listOf(searchInfo.similarRank to SortOrder.DESC, Playlist.avgScore to SortOrder.DESC, Playlist.createdAt to SortOrder.DESC)
+            SearchOrder.Rating -> listOf(Playlist.avgScore to SortOrder.DESC, Playlist.createdAt to SortOrder.DESC)
```

Comment:
Uh yeah... so. Sorting by average score of a playlist means the database needs to know the average score for every playlist. To do that it obviously has to cross join every playlist with every map and aggregate the whole database potentially many times over depending on how many playlists we have.

There are some other issues with the query that exposed is generating (you have to be far more careful to get an efficient query with all these joins) but while those may be fixable this fundamentally isn't. It's why the user table has an upvote column (it's used for sorting the mappers list).

Probably best to include any data we want to sort or filter by into the playlists table. My current list is average score, min nps, max nps, verified and emptyness (could just store the count itself). Honestly at that point there's a fairly good argument for including the rest of the stats too as we're already going to have to have a system to update these.

Random aside: I'm fairly sure the from/to filters are bugged and were meant to filter when the playlist was made not the maps in them.